connectors/src/connectors/webcrawler/temporal/activities.ts

```diff
@@ -141,11 +141,37 @@ export async function crawlWebsiteByConnectorId(connectorId: ModelId) {
   );
 
   await crawler.run([webCrawlerConfig.url]);
+  await crawler.teardown();
+
+  // Adjusting page parent urls.
+  // If the path https://example.com/articles is both a web page and a folder (eg: https://example.com/articles/1),
+  // the WebCrawlerPage object for this page should have its parentUrl set to https://example.com/articles (instead of the default https://example.com/).
+  // This is because we want pages to show up in their folder in the UI.
+  // This can only be done once we know all the folders we have for a given WebConfiguration.url, which is why we do it after the crawlling process.
+  const pages = await WebCrawlerPage.findAll({
+    where: {
+      connectorId: connector.id,
+      webcrawlerConfigurationId: webCrawlerConfig.id,
+    },
+  });
+  for (const page of pages) {
+    const folder = await WebCrawlerFolder.findOne({
+      where: {
+        connectorId: connector.id,
+        webcrawlerConfigurationId: webCrawlerConfig.id,
+        url: normalizeFolderUrl(page.url),
```

Comment:
If you're calling it here we would expect you to call the same normalization code at the moment you insert the folder?